core/templates/dev/head/domain/exploration/OutcomeObjectFactory.js

```diff
@@ -34,17 +34,16 @@ oppia.factory('OutcomeObjectFactory', [
         param_changes: this.paramChanges
       };
     };
-
     /** Returns true iff an outcome has a self-loop and no feedback. */
     Outcome.prototype.isConfusing = function(currentStateName) {
       return (
         this.dest === currentStateName &&
-        !this.feedback.getHtml().trim().length > 0
+        !this.hasNonemptyFeedback()
       );
     };
 
     Outcome.prototype.hasNonemptyFeedback = function() {
-      return !this.feedback.isEmpty();
+      return this.feedback.getHtml().trim() !== '';
```

Comment:
Oh, whoops. I meant, feedback.isEmpty() should be defined as `return feedback.getHtml().trim() !== ''` and this should remain as `return !this.feedback.isEmpty()`.